tests/components/unifiprotect/test_init.py

```diff
@@ -0,0 +1,225 @@
+"""Test the UniFi Protect setup flow."""
+from __future__ import annotations
+
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+from pyunifiprotect import NotAuthorized, NvrError
+
+from homeassistant.components.unifiprotect import async_setup_entry, async_unload_entry
```

Comment:
https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Don't call this directly. Use the standard methods in hass.config_entries and then assert the state of the config entry